Sources/HAP/Server/Server.swift

```diff
@@ -10,148 +11,99 @@ import func Evergreen.getLogger
 
 fileprivate let logger = getLogger("hap.http")
 
+enum ServerError: Error {
+    case couldNotBindToPort(Error)
+    case couldNotStop(Error)
+}
+
 public class Server: NSObject, NetServiceDelegate {
     let device: Device
-    let application: Application
 
     let service: NetService
 
-    let listenSocket: Socket
-    let connectionsLockQueue = DispatchQueue(label: "hap.connectionsLockQueue")
+    let group: EventLoopGroup
+    let bootstrap: ServerBootstrap
 
-    let continueRunningLock = DispatchSemaphore(value: 1)
+    let channel: Channel
 
-    // swiftlint:disable:next identifier_name
-    var _continueRunning = false
-    var continueRunning: Bool {
-        get {
-            continueRunningLock.wait()
-            defer {
-                continueRunningLock.signal()
-            }
-            return _continueRunning
-        }
-        set {
-            continueRunningLock.wait()
-            _continueRunning = newValue
-            continueRunningLock.signal()
-        }
-
-    }
-    var connections = [Int32: Connection]()
-
-    public init(device: Device, port: Int = 0) throws {
+    public init(device: Device, listenPort requestedPort: Int = 0) throws {
         precondition(device.server == nil, "Device already assigned to other Server instance")
         self.device = device
 
-        application = root(device: device)
+        device.controllerHandler = ControllerHandler()
+        device.controllerHandler!.removeSubscriptions = device.removeSubscriberForAllCharacteristics
+
+        let applicationHandler = ApplicationHandler(responder: root(device: device))
+
+        group = MultiThreadedEventLoopGroup(numberOfThreads: System.coreCount)
```

Comment:
I would suggest to only use one thread and thus avoid the need for thread safe code. At least make the number of threads selectable by a function argument. Many use cases will have low traffic and no need for more then one thread.

Reply:
Yes, good idea.